src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -37,21 +41,17 @@ OpBase *NewNodeByLabelScanOp(GraphContext *gc, Node *node) {
     return (OpBase*)nodeByLabelScan;
 }
 
-OpResult NodeByLabelScanConsume(OpBase *opBase, Record *r) {
+OpResult NodeByLabelScanConsume(OpBase *opBase, Record r) {
     NodeByLabelScan *op = (NodeByLabelScan*)opBase;
 
     GrB_Index nodeId;
 
-    // First call to consume.
-    if(ENTITY_GET_ID(op->node) == INVALID_ENTITY_ID) {
-        Record_AddEntry(r, op->node->alias, SI_PtrVal(op->node));
-    }
-
     bool depleted = false;
     GxB_MatrixTupleIter_next(op->iter, NULL, &nodeId, &depleted);
     if(depleted) return OP_DEPLETED;
-
-    Graph_GetNode(op->g, nodeId, op->node);
+    
+    Node *n = Record_GetNode(r, op->node_rec_idx);
+    Graph_GetNode(op->g, nodeId, n);
```

Comment:
Maybe add a comment here to indicate that `Graph_GetNode` is writing into the Record.

Reply:
👍